### PR TITLE
Youtube Integration

### PR DESCRIPTION
**Pull request by oneroyalace**
This PR, together with https://github.com/oneroyalace/YoutubeArchiver and https://github.com/TechAndCheck/hypatia/pull/19, adds the ability to archive YouTube videos. 

To avoid name aliasing issues, the content created by a YouTube channel is referred to as a `YoutubePost` in Zenodotus, while the video featured in that content is referred to as a `YoutubeVideo`.

Oh! Open request for a new name to replace `YoutubeArchiver`. Please!

# New environment variablers
- `HYPATIA_SERVER_URL`
- `HYPATIA_AUTH_KEY`

# Testing
1. `bundle install`
2. `rails db:migrate`
3. `rails t`

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -136,6 +136,7 @@ gem "streamio-ffmpeg"
 # gem "zorki", "0.1.0", git: "https://github.com/cguess/zorki"
 gem "birdsong", "0.1.0", git: "https://github.com/cguess/birdsong"
 gem "forki", git: "https://github.com/oneroyalace/forki"
+gem "youtubearchiver", path: "~/hub/YoutubeArchiver/"
```
Comment (cguess) [inline Gemfile:139]: This shouldn't be necessary for zenodotus? We should also be removing the others honestly.

File: app/controllers/text_search_controller.rb [hunk 2]
```diff
@@ -28,7 +28,7 @@ def search
     @post_search_hits = []
     @user_search_hits = []
 
-    post_models = [Sources::Tweet, Sources::InstagramPost, Sources::FacebookPost]
+    post_models = [Sources::Tweet, Sources::InstagramPost, Sources::FacebookPost, Sources::YoutubePost]
```
Comment (cguess) [inline app/controllers/text_search_controller.rb:31]: Should we rename this to `YouTube` to accurate represent the name of the project?

**Review by cguess — COMMENTED** [review]
Going to wait to finish this until Hypatia is passing.

File: Gemfile (hunk 1 above)
Comment (oneroyalace) [reply Gemfile:139]: I think they're used for Sorbet type checking

File: Gemfile [hunk 3]
```diff
@@ -136,6 +136,9 @@ gem "streamio-ffmpeg"
...
```
Comment (cguess) [inline Gemfile:139]: Breaks this :-)

Comment (oneroyalace) [reply Gemfile:139]: Ah shoot.

**Comment by cguess** [comment]
Tests are failing with
```
Error:
YoutubeChannelTest#test_can_update_youtube_channel:
NoMethodError: undefined method `<<' for nil:NilClass

          base_url << '?'
                   ^^
    app/media_sources/youtube_media_source.rb:88:in `retrieve_youtube_post!'
    app/media_sources/youtube_media_source.rb:23:in `extract'
    test/models/youtube_channel_test.rb:7:in `setup'
```

File: app/views/archive/_list_item_youtube_post.html.erb [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        </div>
+        <div class="sm:w-2/3 sm:pl-8 sm:py-8 sm:border-l border-gray-200 sm:border-t-0 border-t mt-4 pt-4 sm:mt-0 text-center sm:text-left">
+
+          <%= render partial: "item_media_preview", locals: { item: archive_item.youtube_post } %>
+
+          <p class="leading-relaxed text-lg mb-4">
+            <%= archive_item.youtube_post.title %>
+          </p>
+          <p class="text-base"><%= archive_item.youtube_post.posted_at %></p>
+          <!-- <a class="text-indigo-500 inline-flex items-center">Learn More
```
Comment (cguess) [inline app/views/archive/_list_item_youtube_post.html.erb:31]: Mind deleting this out?

**Review by cguess — APPROVED** [review]
One little thing, rest is good!

**Pull request merged**